src/instance.ts

```diff
@@ -193,6 +249,269 @@ class Instance extends common.GrpcServiceObject {
+   *     automatically.
+   * @property {number} [maxApiCalls] Maximum number of API calls to make.
+   * @property {number} [maxResults] Maximum number of items to return.
+   * @property {number} [pageSize] Maximum number of results per page.
+   * @property {string} [pageToken] A previously-returned page token
+   *     representing part of the larger set of results to view.
+   */
+  /**
+   * @typedef {array} GetBackupOperationsResponse
+   * @property {IOperation[]} 0 Array of {@link IOperation} instances.
+   * @property {object} 1 The full API response.
+   */
+  /**
+   * List pending and completed backup operations for all databases in the instance.
+   *
+   * @see {@link #listOperations}
+   *
+   * @param {GetBackupOperationsOptions} [options] The query object for listing
+   *     backup operations.
+   * @param {CallOptions} [options.gaxOptions] The request configuration
+   *     options, outlined here:
+   *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+   * @returns {Promise<GetBackupOperationsResponse>} when resolved, contains a
+   *     paged list of backup operations.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const [operations] = await instance.getBackupOperations();
+   */
+  getBackupOperations(
+    optionsOrCallback?:
+      | GetBackupOperationsOptions
+      | GetBackupOperationsCallback,
+    cb?: GetBackupOperationsCallback
+  ): void | Promise<GetBackupOperationsResponse> {
+    const callback =
+      typeof optionsOrCallback === 'function'
+        ? (optionsOrCallback as GetBackupOperationsCallback)
+        : cb!;
+    const options =
+      typeof optionsOrCallback === 'object'
+        ? (optionsOrCallback as GetBackupOperationsOptions)
+        : {gaxOptions: {}};
+    const gaxOpts: CallOptions = options.gaxOptions as CallOptions;
+    const reqOpts = extend({}, options, {
+      parent: this.formattedName_,
+    });
+    delete reqOpts.gaxOptions;
+
+    this.request<
+      IOperation,
+      databaseAdmin.spanner.admin.database.v1.IListBackupOperationsResponse
+    >(
+      {
+        client: 'DatabaseAdminClient',
+        method: 'listBackupOperations',
+        reqOpts,
+        gaxOpts,
+      },
+      (err, operations, ...args) => {
+        callback!(err, operations, ...args);
+      }
+    );
+  }
+
+  getDatabaseOperations(
+    options?: GetDatabaseOperationsOptions
+  ): Promise<GetDatabaseOperationsResponse>;
+  getDatabaseOperations(callback: GetDatabaseOperationsCallback): void;
+  getDatabaseOperations(
+    options: GetDatabaseOperationsOptions,
+    callback: GetDatabaseOperationsCallback
+  ): void;
+  /**
+   * Query object for listing database operations.
+   *
+   * @typedef {object} GetDatabaseOperationsRequest
+   * @property {boolean} [autoPaginate=true] Have pagination handled
```

Comment:
Do we need to specially handle this option if provided by a user, to box it in "gaxOptions" for them?

Reply:
Ah this goes back to an earlier reply I made in src/database.ts. Since we have GetDatabaseOperationsOptions, we no longer need this type so I've now removed it. Users can provide autoPaginate in gaxOptions.

Reply:
I like that option from a separation of responsibilities standpoint, but I think this is our first deviation from just accepting that option from our library, without the user knowing what this "gax" thing is. It appears quite a bit in other methods as well: https://github.com/googleapis/nodejs-spanner/search?q=autopaginate&unscoped_q=autopaginate. We could run this search for any other library as well, and it would be a top-level option.

I think even though those methods accept it as a top-level option, it isn't actually having the intended effect, as we aren't passing it to gax for them via 'gaxOptions'.

So I think we have a couple of options, that I can think of:

1. Continue accepting autoPaginate as if it's our own option, and pass it as  "gaxOptions.autoPaginate" for the user
  - We would want to fix the other places we aren't doing this already
1. Rid ourselves of this responsibility-- that's "gax" stuff, so the user will have to pass it through "gaxOptions"
  - We would need to change our examples to show it as "gaxOptions"

What do you think?

Reply:
I'm a bit torn here. On the one hand, I don't want us to deviate too much from existing code. But then again, creating custom types with CallOptions is not sustainable since then we have to remember to update the types any time a call option is added.

So I'm in favor of option 2.

In the case of existing code, I think some of it actually does work. E.g. Instance#getDatabases()
https://github.com/googleapis/nodejs-spanner/blob/568a244dc03854ff120fa943930bf4fb2e25a927/src/instance.ts#L699

This is because it passes the whole input to gaxOpts and I'm assuming the auto-generated code is somehow able to extract the relevant bits.

Let me try to fix the examples in the code that we've added in this PR.

Reply:
I've now added examples on how to setup autoPaginate to all the get*s() methods we've introduced in this PR.